dbms/src/DataTypes/DataTypeTuple.cpp

```diff
@@ -188,6 +191,12 @@ void DataTypeTuple::deserializeText(IColumn & column, ReadBuffer & istr, const F
         }
     });
 
+    // Special format for one element tuple (1,)
+    if (1 == elems.size())
+    {
+        skipWhitespaceIfAny(istr);
+        assertChar(',', istr);
```

Comment:
It still cannot parse one element tuples without extra comma at the end, though previous versions don't add extra comma. Possibly, it will be better just skip extra comma (if there is one), not assert it

Reply:
Yup, it sounds like a good backward-compatible decision.